src/main/java/org/littleshoot/proxy/mitm/BouncyCastleSslEngineSource.java

```diff
@@ -148,8 +148,7 @@ public BouncyCastleSslEngineSource(Authority authority,
     private void filterWeakCipherSuites(SSLEngine sslEngine) {
         List<String> ciphers = new LinkedList<String>();
         for (String each : sslEngine.getEnabledCipherSuites()) {
-            if (each.equals(each.equals("TLS_DHE_RSA_WITH_AES_128_CBC_SHA")
-                    || each.equals("TLS_DHE_RSA_WITH_AES_256_CBC_SHA"))) {
+            if (each.equals("TLS_DHE_RSA_WITH_AES_128_CBC_SHA") || each.equals("TLS_DHE_RSA_WITH_AES_256_CBC_SHA")) {
```

Comment:
This is a great catch! Duplicate `equals(each.equals(` -> wrong condition, -> missing test

 squid:S2159 - Silly equality checks should not be made.